bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/NoboHubBridgeHandler.java

```diff
@@ -0,0 +1,415 @@
+                ht.start();
+                hubThread = ht;
+
+                if (ht.getConnection().isConnected()) {
+                    logger.debug("Communication thread to {} is up and running, we are online", hostName);
+                    updateProperty(Thing.PROPERTY_SERIAL_NUMBER, serialNumber);
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    logger.debug("HubCommunicationThread is not connected anymore, setting to OFFLINE");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/message.bridge.connection.failed");
+                }
+            } catch (NoboCommunicationException commEx) {
+                logger.debug("HubCommunicationThread failed, exiting thread");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, commEx.getMessage());
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing NoboHub '{}'", getThing().getUID().getId());
+
+        final NoboThingDiscoveryService discoveryService = this.discoveryService;
+        if (discoveryService != null) {
+            discoveryService.stopScan();
+        }
+
+        HubCommunicationThread ht = this.hubThread;
+        if (ht != null) {
+            logger.debug("Stopping communication thread");
+            ht.stopNow();
+        }
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler handler, Thing thing) {
+        logger.info("Adding thing: {}", thing.getLabel());
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler handler, Thing thing) {
+        logger.info("Disposing thing: {}", thing.getLabel());
+    }
+
+    private void onUpdate(Hub hub) {
+        logger.debug("Updating Hub: {}", hub.getName());
+        this.hub = hub;
+        OverridePlan activeOverridePlan = getOverride(hub.getActiveOverrideId());
+
+        if (null != activeOverridePlan) {
+            logger.debug("Updating Hub with ActiveOverrideId {} with Name {}", activeOverridePlan.getId(),
+                    activeOverridePlan.getMode().name());
+
+            updateState(NoboHubBindingConstants.CHANNEL_HUB_ACTIVE_OVERRIDE_NAME,
+                    StringType.valueOf(activeOverridePlan.getMode().name()));
+        }
+
+        // Update all zones to set online status and update profile name from weekProfileRegister
+        for (Zone zone : zoneRegister.values()) {
+            refreshZone(zone);
+        }
+
+        updateProperty(PROPERTY_HOSTNAME, hub.getName());
+        updateProperty(Thing.PROPERTY_SERIAL_NUMBER, hub.getSerialNumber().toString());
+        updateProperty(PROPERTY_SOFTWARE_VERSION, hub.getSoftwareVersion());
+        updateProperty(Thing.PROPERTY_HARDWARE_VERSION, hub.getHardwareVersion());
+        updateProperty(PROPERTY_PRODUCTION_DATE, hub.getProductionDate());
```

Comment:
`editProperties()` should be used since multiple properties are being updated:
https://www.openhab.org/javadoc/latest/org/openhab/core/thing/binding/basethinghandler#updateProperty(java.lang.String,java.lang.String)